path_roles.go

```diff
@@ -200,21 +232,12 @@ func setTerraformRole(ctx context.Context, s logical.Storage, name string, roleE
 	return nil
 }
 
-func saveRole(ctx context.Context, s logical.Storage, c *terraformRoleEntry, name string) error {
-	entry, err := logical.StorageEntryJSON("roles/"+name, c)
-	if err != nil {
-		return err
-	}
-
-	return s.Put(ctx, entry)
-}
-
-func getRole(ctx context.Context, s logical.Storage, name string) (*terraformRoleEntry, error) {
+func (b *tfBackend) getRole(ctx context.Context, s logical.Storage, name string) (*terraformRoleEntry, error) {
 	if name == "" {
-		return nil, errwrap.Wrapf("missing role name", nil)
+		return nil, fmt.Errorf("missing role name")
 	}
 
-	entry, err := s.Get(ctx, "roles/"+name)
+	entry, err := s.Get(ctx, "role/"+name)
```

Comment:
We might have to add an upgrade path for this since entries were previously saved under `roles/<name>`.

Reply:
Update: This is coming out of a PoC-ish state, so we don't have to worry about storage upgrades.